### PR TITLE
British Columbia Pension Income Amount

### LINKED ISSUE #470: British Columbia Pension Income Amount
Reference: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1bc/td1bc-23e.pdf

### PR DESCRIPTION
**Pull request by YiweiJ**
Fixes #470

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/pension_income_amount/cap.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+description: British Columbia provides this maximum pension benefit.
```
Comment (Yaohhhh) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/pension_income_amount/cap.yaml:1]: description: British Columbia provides this max pension benefit for eligible filers.

File: policyengine_canada/tests/gov/provinces/bc/tax/income/credits/bc_pension_income_amount/bc_pension_income_amount.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+- name: Max Pension Benefit
```
Comment (Yaohhhh) [inline policyengine_canada/tests/gov/provinces/bc/tax/income/credits/bc_pension_income_amount/bc_pension_income_amount.yaml:1]: - name: Pension income is larger than the maximum amount

File: policyengine_canada/tests/gov/provinces/bc/tax/income/credits/bc_pension_income_amount/bc_pension_income_amount.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+- name: Max Pension Benefit 
+  period: 2023
+  input:
+    province_code: BC
+    pension_and_savings_plan_income: 1_001
+  output:
+    bc_pension_income_amount: 1_000
+
+- name: Pension Benefit Below Maximum Amount
```
Comment (Yaohhhh) [inline policyengine_canada/tests/gov/provinces/bc/tax/income/credits/bc_pension_income_amount/bc_pension_income_amount.yaml:9]: - name: Pension income is less than the maximum amount

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/non_refundable.yaml [hunk 4]
```diff
@@ -3,6 +3,7 @@ values:
   2022-01-01:
     - bc_tax_reduction_credit
     - bc_age_credit
+    - bc_pension_income_amount
 
 metadata:
   unit: program
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/non_refundable.yaml:9]: the metadata will be fixed in the BC basic personal amount

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**